src/NLog/Layouts/LayoutParser.cs

```diff
@@ -386,7 +386,10 @@ private static LayoutRenderer ParseLayoutRenderer(ConfigurationItemFactory confi
                         var value = ParseParameterValue(stringReader);
                         if (!string.IsNullOrEmpty(parameterName) || !StringHelpers.IsNullOrWhiteSpace(value))
                         {
-                            // TODO NLog 5.0 Should throw exception when invalid configuration (Check throwConfigExceptions)
+                            if (throwConfigExceptions ?? LogManager.ThrowConfigExceptions == true)
+                            {
+                                throw new NLogConfigurationException("Unrecognized property '{0}={1}` for ${{{2}}} ({3})", parameterName, value, name, layoutRenderer?.GetType());
```

Comment:
Instead of having the same log formatting twice, then one could consider this:

```c#
var configException = new NLogConfigurationException("Unrecognized property '{0}={1}` for ${{{2}}} ({3})", parameterName, value, name, layoutRenderer?.GetType());
if (throwConfigExceptions ?? LogManager.ThrowConfigExceptions ?? LogManager.ThrowExceptions)
     throw configException;

InternalLogger.Warn(configException.Message);
```

Reply:
This makes `WhenEmpty_MissingInner_ShouldNotThrow` fail as well.

Reply:
Can you show why it fails?